src/dev-guide/system-architecture.md

```diff
@@ -0,0 +1,50 @@
+# System Architecture
+
+## Controller-Service-Repository
+
+The architecture for this project follows a controller-service-repository pattern which provides strong separation between the business logic, and database logic. This allows each component of the frontend and backend to be self-contained, have reduced coupling, and increased extendability.
+
+This is implemented by the backend providing REST APIs for the frontend in the controller layer. These APIs use business logic and functionality from the service layer, which manage the data in the database with the repository layer. The separation of the logics are described in the following diagram:
+
+
+![](file:///C:/Users/Judy/AppData/Local/Temp/msohtmlclip1/01/clip_image002.jpg)![1666253180277](image/system-archtiecture/1666253180277.png)
+
+Source: [https://tom-collings.medium.com/controller-service-repository-16e29a4684e5](https://tom-collings.medium.com/controller-service-repository-16e29a4684e5)
```

Comment:
Alt text (the bit in square brackets) should contain things that describe the image for visually impaired people.